site/src/common/hooks/usePagination.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react'
+import { NumberParam, useQueryParam } from 'use-query-params'
+import { ITEMS_PER_PAGE } from '../constants'
+
+
+export const usePagination = <TItem extends {}>(list: Array<TItem>): {
```

Comment:
Can you create interface Pagination(hasNext, next,  prev, items, pagesCount, setPage) ?